File: arduinobot_cpp_examples/src/simple_service_server.cpp
#include <rclcpp/rclcpp.hpp>
#include <memory>
#include <arduinobot_msgs/srv/add_two_ints.hpp>

class SimpleServiceServer: public rclcpp::Node{

    private:
        rclcpp::Service<arduinobot_msgs::srv::AddTwoInts>::SharedPtr service_;

        void serviceCallback(
            const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Request> request,
            const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Response> response
        ){
            RCLCPP_INFO(this->get_logger(), "Incoming request\na: %ld   " "b: %ld", request->a, request->b);
            response->sum = request->a + request->b;
            RCLCPP_INFO(this->get_logger(), "Sending back response: [%ld]", response->sum);
        }

    public:
        SimpleServiceServer(): Node("simple_service_server"){
            service_ = create_service<arduinobot_msgs::srv::AddTwoInts>(
                "add_two_ints", std::bind(&SimpleServiceServer::serviceCallback, this, std::placeholders::_1, std::placeholders::_2)
            );

            RCLCPP_INFO(this->get_logger(), "Service add_two_ints started");

        }
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleServiceServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: arduinobot_py_examples/arduinobot_py_examples/simple_service_server.py
import rclpy
from rclpy.node import Node
from arduinobot_msgs.srv import AddTwoInts

class SimpleServiceServer(Node):
    def __init__(self, name):
        super().__init__(name)

        # Create a service
        self.service = self.create_service(AddTwoInts, "add_two_ints", self.service_callback)
        # Logger message
        self.get_logger().info("Service Server is ready")

    def service_callback(self, request, response):
        response.sum = request.a + request.b 
        self.get_logger().info("Incoming request\na: %d b: %d" % (request.a, request.b))
        self.get_logger().info("Sending response: %d" % response.sum)
        return response
    
def main(args=None):
    rclpy.init(args=args)
    node = SimpleServiceServer("simple_service_server")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()